File: 第十三章_智力题/数组清空练习题/Clear.cpp
class Clear {
public:
    int getWinner(vector<int> A, int n) {
        sort(A.begin(),A.end());
        int res=0;
        int j;
        for(int i=0;i<n;){
            int num=0;
            j=i;
            while(j<n&&A[i]==A[j]){
                j++;
                num++;
            }
            i=j;
            res^=num;
        }
        return res!=0;
    }
};

File: 第十三章_智力题/数组清空练习题/Clear.java
import java.util.*;

public class Clear {
    public int getWinner(int[] A, int n) {
        // write code here
        int[] count=new int[n+1];
        for(int i=0;i<n;i++){
            count[A[i]]++;
        }
        int res=0;
        for(int i=1;i<=n;i++){
            res^=count[i];
        }
        return res!=0?1:0;
    }
}